projects/cashmere/src/lib/tabs/tab-set/tab-set.component.ts

```diff
@@ -70,7 +82,11 @@ export class TabSetComponent implements AfterContentInit {
         }
     }
 
-    constructor(private router: Router, private route: ActivatedRoute) {}
+    constructor(private router: Router, private route: ActivatedRoute, @Self() private _tabsService: TabsService) {}
```

Comment:
What is the reasoning for introducing a service here?

Reply:
I wanted to be able to tell the TabComponent children what their parent's direction was without having to initialize the content of the TabSetComponent first. This way the TabComponent's direction would be correct before it ever initializes its view or content. If there's a better way to do this I'm all ears, I just read a service was good practice for parent-child communication.

Reply:
Services are Singleton in Angular; what if you have multiple tab sets, or nested tabs?

Reply:
I believe it should create a new TabsService for every TabSetComponent since the TabSetComponent lists "TabsService" under "providers" in the @Component decorator. The @Self decorator in the constructor should also prevent TabSetComponent from getting the dependency from anywhere other than its own injector. Each TabComponent should get the dependency from their parents because they don't provide it themselves and the next place to look would be their parent component.